sound/pci/hda/patch_hdmi.c

```diff
@@ -1703,30 +1767,36 @@ static void silent_stream_enable(struct hda_codec *codec,
 	/* configure unused pins to choose other converters */
 	pin_cvt_fixup(codec, per_pin, 0);
 
-	snd_hdac_sync_audio_rate(&codec->core, per_pin->pin_nid,
-				 per_pin->dev_id, I915_SILENT_RATE);
-
-	/* trigger silent stream generation in hw */
-	format = snd_hdac_calc_stream_format(I915_SILENT_RATE, I915_SILENT_CHANNELS,
-					     I915_SILENT_FORMAT, I915_SILENT_FORMAT_BITS, 0);
-	snd_hda_codec_setup_stream(codec, per_pin->cvt_nid,
-				   I915_SILENT_FMT_MASK, I915_SILENT_FMT_MASK, format);
-	usleep_range(100, 200);
-	snd_hda_codec_setup_stream(codec, per_pin->cvt_nid, I915_SILENT_FMT_MASK, 0, format);
-
-	per_pin->channels = I915_SILENT_CHANNELS;
-	hdmi_setup_audio_infoframe(codec, per_pin, per_pin->non_pcm);
+	if (spec->silent_stream_type == SILENT_STREAM_I915)
+		silent_stream_enable_i915(codec, per_pin);
+	else
+		silent_stream_enable_kae(codec, per_pin);
```

Comment:
this is odd, the silent_stream_type can have three values, so you would enable_kae() even when the silent_stream_type is SILENT_STREAM_OFF ??

Reply:
We don't call this function unless silent-streaming is enabled (i.e. value can never be SILENT_STREAM_OFF if we reach this part of code).

Reply:
@plbossart  Updated now with hopefully more understandable flow. I also added a piece of text to the Kconfig option to explain the difference.